File: backend/src/main/java/com/todo1/hulkstore/domain/Product.java
package com.todo1.hulkstore.domain;

import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.FieldDefaults;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;
import java.math.BigDecimal;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@FieldDefaults(level = AccessLevel.PRIVATE)
@Entity
@Table(name = "product")
public class Product {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Long id;

    String name;

    @ManyToOne()
    @JoinColumn(name = "product_type_id", nullable = false)
    ProductType productType;

    BigDecimal price;

    Integer stock;
}

File: frontend/src/app/products/models/product.ts
import { ProductType } from './product-type';

export interface Product {
  id: number;
  name: string;
  productType: ProductType;
  productTypeName: string;
  stock: number;
  price: number;

  isSelected?: boolean;
}

export interface ProductRequest {
  id?: number;
  name: string;
  productTypeId: number;
  stock: number;
  price: number;
}

export enum ProductStockCondition {
  ALL = 'all',
  AVAILABLE = 'available',
  UNAVAILABLE = 'unavailable'
}
